lib/std/crypto/kyber.zig

```diff
@@ -1466,49 +1429,33 @@ fn Mat(comptime K: u8) type {
     };
 }
 
-// Returns 0 if a = b and 1 otherwise.
-fn cteq(comptime len: usize, a: *const [len]u8, b: *const [len]u8) usize {
-    // TODO Is there an existing function in the stdlib for this?
-    var i: usize = 0;
-    var ret: u8 = 0;
-    while (i < len) : (i += 1) {
-        ret |= a[i] ^ b[i];
-    }
-    return @bitCast(usize, -@as(isize, ret)) >> (@typeInfo(usize).Int.bits - 1);
+// Returns `true` if a ≠ b.
+fn ctneq(comptime len: usize, a: [len]u8, b: [len]u8) bool {
```

Comment:
Using `bool` here scares me. Wouldn't the compiler be more tempted to introduce a branch on the two possible values?

Reply:
`bool` is a `u1` in disguise. Inverting it just causes the compiled to emit a XOR.

Reply:
When you pass the bool to `cmov`, will the compiler think it's a good idea to specialise the function for the true and false cases? I don't know. This is certainly not a hard science and it'd be best if we had a `secret` attribute to decorate the bool with.

Reply:
A `secret` attribute is something that's been discussed for a years with the LLVM folks, but this is unfortunately unlikely to ever get implemented.

With `cmov` in the standard library, we'll have the opportunity to use assembly or at least check that the compiler does what we expect. We know exactly what compiler is used with the standard library, so, compared to C, it helps a lot!

In the meantime, I'm going to change this to a `u1` if that makes you feel more comfortable :)

Reply:
No more `bool`.

Reply:
Writing this in assembly seems the way to go, but I suppose that's well out of scope of this PR. :)